src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -94,6 +100,37 @@ func (r *StatefulSetResource) runUpdate(
 	return nil
 }
 
+func (r *StatefulSetResource) isClusterHealthy(ctx context.Context) error {
```

Comment:
afaiu in #3023, at this point the cluster should be in maintenance mode. should we enforce it and add a check here?

Reply:
In our current rolling update we